bmwqemu.pm

```diff
@@ -108,6 +108,21 @@ our $testedversion;
 
 sub init {
     load_vars();
+    {
+        # DEBUG in vars.json can be a single value or a list (array) of values.
+        # it's turned into a hash which should be tested using exists, like so:
+        # if (exists get_var("DEBUG")->{flag}) { ... }
+        # this allows to give a combination of flags
+        my %DEBUG_SET;
+        if (exists $vars{DEBUG}) {
+            my $DEBUG = \$vars{DEBUG};
+            if (ref $DEBUG eq "SCALAR") {
+                $DEBUG = [$$DEBUG];
+            }
+            @DEBUG_SET{@$$DEBUG} = ();
+        }
+        $vars{DEBUG} = \%DEBUG_SET;
+    };
     $vars{NAME} ||= 'noname';
```

Comment:
if you start it for the first time, I get this in vars.json:

 "DEBUG" : {},

if I start it another time, I get:

Not an ARRAY reference at /suse/coolo/prod/os-autoinst/bmwqemu.pm line 122.

Less is more - please make DEBUG a string of flags seperated by space instead of this fanciness